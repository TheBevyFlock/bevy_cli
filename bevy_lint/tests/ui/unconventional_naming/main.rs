#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::unconventional_naming)]
//~^ NOTE: the lint level is defined here
#![allow(dead_code)]
use bevy::prelude::*;

//~v NOTE: `SystemSet` implemented here
#[derive(SystemSet, Debug, Clone, PartialEq, Eq, Hash)]
struct MyAudio;
//~^ ERROR: Unconventional struct name for this trait impl
//~| HELP: rename the SystemSet
