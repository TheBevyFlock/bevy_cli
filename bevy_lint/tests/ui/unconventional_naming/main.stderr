error: unconventional type name for a `Plugin` or `SystemSet`
  --> tests/ui/unconventional_naming/main.rs:10:8
   |
10 | struct MyAudio;
   |        ^^^^^^^ help: rename the SystemSet: `MyAudioSet`
   |
note: `SystemSet` implemented here
  --> tests/ui/unconventional_naming/main.rs:9:10
   |
9  | #[derive(SystemSet, Debug, Clone, PartialEq, Eq, Hash)]
   |          ^^^^^^^^^
note: the lint level is defined here
  --> tests/ui/unconventional_naming/main.rs:3:9
   |
3  | #![deny(bevy::unconventional_naming)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in the derive macro `SystemSet` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unconventional type name for a `Plugin` or `SystemSet`
  --> tests/ui/unconventional_naming/main.rs:15:8
   |
15 | struct Foo;
   |        ^^^ help: rename the Plugin: `FooPlugin`
   |
note: `Plugin` implemented here
  --> tests/ui/unconventional_naming/main.rs:20:1
   |
20 | / impl Plugin for Foo {
21 | |     fn build(&self, _app: &mut App) {}
22 | | }
   | |_^

error: aborting due to 2 previous errors

